src/super_gradients/training/datasets/obb/dota.py

```diff
@@ -0,0 +1,340 @@
+from super_gradients.common.registry import register_dataset
+from super_gradients.dataset_interfaces import HasClassesInformation
+from super_gradients.training.transforms import OBBDetectionCompose
+from super_gradients.training.transforms.obb import OBBSample
+from torch.utils.data import Dataset
+from super_gradients.common.factories.transforms_factory import TransformsFactory
+
+__all__ = ["DOTAOBBDataset"]
+
+
+@register_dataset()
+class DOTAOBBDataset(Dataset, HasPreprocessingParams, HasClassesInformation):
+    @resolve_param("transforms", TransformsFactory())
+    def __init__(
+        self,
+        data_dir,
+        transforms,
+        class_names: Iterable[str],
+        ignore_empty_annotations: bool = False,
+        difficult_labels_are_crowd: bool = False,
+        images_ext: str = ".jpg",
+        images_subdir="images",
+        ann_subdir="ann-obb",
+    ):
+        super().__init__()
+
+        images_dir = Path(data_dir) / images_subdir
+        ann_dir = Path(data_dir) / ann_subdir
+        images, labels = self.find_images_and_labels(images_dir, ann_dir, images_ext)
+        self.images = []
+        self.coords = []
+        self.classes = []
+        self.difficult = []
+        self.transforms = OBBDetectionCompose(transforms, load_sample_fn=self.load_random_sample)
+        self.class_names = list(class_names)
+        self.difficult_labels_are_crowd = difficult_labels_are_crowd
+
+        class_names_to_index = {name: i for i, name in enumerate(self.class_names)}
+        for image_path, label_path in tqdm(zip(images, labels), desc=f"Parsing annotations in {ann_dir}", total=len(images)):
+            coords, classes, difficult = self.parse_annotation_file(label_path)
+            if ignore_empty_annotations and len(coords) == 0:
+                continue
+            self.images.append(image_path)
+            self.coords.append(coords)
+            self.classes.append(np.array([class_names_to_index[c] for c in classes], dtype=int))
+            self.difficult.append(difficult)
+
+    def __len__(self):
+        return len(self.images)
+
+    def load_random_sample(self) -> OBBSample:
+        num_samples = len(self)
+        random_index = random.randrange(0, num_samples)
+        return self.load_sample(random_index)
+
+    def load_sample(self, index) -> OBBSample:
+        image = cv2.imread(str(self.images[index]))
+        coords = self.coords[index]
+        classes = self.classes[index]
+        difficult = self.difficult[index]
+
+        cxcywhr = poly_to_cxcywhr(coords)
+
+        is_crowd = difficult.reshape(-1) if self.difficult_labels_are_crowd else np.zeros_like(difficult, dtype=bool)
+        sample = OBBSample(
+            image=image,
+            rboxes_cxcywhr=cxcywhr.reshape(-1, 5),
+            labels=classes.reshape(-1),
+            is_crowd=is_crowd,
+        )
+        return sample.sanitize_sample()
+
+    def __getitem__(self, index) -> OBBSample:
```

Comment:
This is not consistent with how we work in DetectionDataset, where we return tensors and the collate works on them.
This is fine of course, but I would like to understand why, since like I stated earlier I didn't understand why we need to untroduce new collate.